pkg/cmd/project/list/list.go

```diff
@@ -0,0 +1,233 @@
+package list
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/cli/cli/v2/pkg/cmdutil"
+
+	"github.com/cli/cli/v2/pkg/cmd/project/shared/format"
+	"github.com/cli/cli/v2/pkg/cmd/project/shared/queries"
+
+	"github.com/cli/cli/v2/internal/tableprinter"
+	"github.com/cli/go-gh/v2/pkg/api"
+	"github.com/spf13/cobra"
+)
+
+type listOpts struct {
+	limit     string
+	web       bool
+	userOwner string
+	orgOwner  string
+	closed    bool
+	format    string
+}
+
+type listConfig struct {
+	tp        *tableprinter.TablePrinter
+	client    *api.GraphQLClient
+	opts      listOpts
+	URLOpener func(string) error
+}
+
+func parseLimit(limit string) (int, error) {
+	if limit == "" {
+		return queries.LimitMax, nil
+	} else if limit == "all" {
+		return 0, nil
+	}
+
+	v, err := strconv.Atoi(limit)
+	if err != nil {
+		return 0, fmt.Errorf("invalid value '%s' for limit", limit)
+	}
+	return v, nil
+}
+
+func NewCmdList(f *cmdutil.Factory, runF func(config listConfig) error) *cobra.Command {
+	opts := listOpts{}
+	listCmd := &cobra.Command{
+		Short: "List the projects for an owner",
+		Use:   "list",
+		Example: `
+# list the current user's projects
+gh project list
+
+# open projects for user monalisa in the browser
+gh project list --user monalisa --web
+
+# list the projects for org github including closed projects
+gh project list --org github --closed
+
+# add --format=json to output in JSON format
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := queries.NewClient()
+			if err != nil {
+				return err
+			}
+
+			URLOpener := func(url string) error {
+				return f.Browser.Browse(url)
+			}
+			t := tableprinter.New(f.IOStreams)
+			config := listConfig{
+				tp:        t,
+				client:    client,
+				opts:      opts,
+				URLOpener: URLOpener,
+			}
+
+			// allow testing of the command without actually running it
+			if runF != nil {
+				return runF(config)
+			}
+			return runList(config)
+		},
+	}
+
+	listCmd.Flags().StringVar(&opts.userOwner, "user", "", "Login of the user owner")
+	listCmd.Flags().StringVar(&opts.orgOwner, "org", "", "Login of the organization owner.")
+	listCmd.Flags().BoolVarP(&opts.closed, "closed", "c", false, "Show closed projects.")
+	listCmd.Flags().BoolVarP(&opts.web, "web", "w", false, "Open projects list in the browser.")
+	listCmd.Flags().StringVar(&opts.format, "format", "", "Output format, must be 'json'.")
+	listCmd.Flags().StringVar(&opts.limit, "limit", "", "Maximum number of projects. Defaults to 100. Set to 'all' to list all projects. Note that closed projects are filtered from the final results without the --closed flag.")
```

Comment:
This is a very long flag description, plus:
- We typically allow the shorthand `-L` wherever there is `--limit`
- We deliberately do _not_ allow unlimited listing (i.e. `"all"`). Users are required to pass a very large number to opt into getting many pages of results.
- The note about closed projects is not necessary. I would expect only to see open projects in the list unless I passed `--closed`.

Reply:
addressed in 6c89a5bd

Reply:
I remember now why I added `closed` to the help output (after a bunch of repeat debugging lol). say for example I run the following:

```
project list --org cli --format=json --limit=3 
```

if any of the projects are closed, they are filtered from the list. I have supplied the limit flag of 3, but may anywhere between 0 and 3 results, depending on the order returned by the API. To make this more confusing, the json output has a field `totalCount` which is the sum of open and closed projects, so a user may be confused wondering why we don't return all of the results to them.